collections/collections.go

```diff
@@ -0,0 +1,98 @@
+package collections
+
+import (
+	"bytes"
+	"fmt"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+)
+
+// StorageProvider represents sdk.Context
+// it is used to avoid to reduce dependencies.
+type StorageProvider interface {
```

Comment:
Thought of abstracting sdk.Context as a StorageProvider interface in order not to have to depend on the SDK. 

I've seen there's work to decouple store from the SDK fully, currently I cannot depend only on store due to  dependency on store=>test=>codec

Reply:
Let's do everything we can to avoid a direct dependency on the sdk. Store is being split out currently. There is also the core API proposed to address this

Reply:
Ok, so I assume using this interface should be fine. Feel free to re-open in case there are issues.